networkx/algorithms/similarity.py

```diff
@@ -982,8 +974,9 @@ def get_edit_paths(
             # assert not len(pending_g)
             # assert not len(pending_h)
             # path completed!
-            # assert matched_cost <= maxcost.value
-            maxcost.value = min(maxcost.value, matched_cost)
+            # assert matched_cost <= maxcost_value
+            nonlocal maxcost_value
```

Comment:
`nonlocal` is a code smell. Possibly outside the scope of this PR to address though.